### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by slorek**
## Description
The contributing guidelines document formalises the team's agreed approach to pull requests, and some agreed code conventions. It is intended to be a living document which can be iterated.

I've also updated the checklist template to make it more relevant and we should ensure this is completed or redacted for future changes.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,16 +3,20 @@
...
+<!--- Put an `x` in all the boxes that apply. Delete items which are not relevant. -->
 ## Checklist:
-<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
-- [ ] Automated checks are passing locally.
-- [ ] CHANGELOG updated if change is worth telling users about.
-### General testing
-- [ ] Test without javascript
+- [ ] Have you documented your changes in the description?
+- [ ] Does the change present any security considerations?
+- [ ] Is any gem functionality overloaded?
```
Comment (frankieroberto) [inline .github/PULL_REQUEST_TEMPLATE.md]: This is a bit unclear to me.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -3,16 +3,20 @@
...
 <!--- Describe your changes in detail -->
 
+<!--- Put an `x` in all the boxes that apply. Delete items which are not relevant. -->
 ## Checklist:
-<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
-- [ ] Automated checks are passing locally.
-- [ ] CHANGELOG updated if change is worth telling users about.
-### General testing
-- [ ] Test without javascript
+- [ ] Have you documented your changes in the description?
```
Comment (frankieroberto) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Have you documented your changes in the pull request description?
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Contributing
+This project is made available as open source, and we welcome contributions from the public. The Department also employs a team dedicated to improving the product. The product team are the maintainers of the codebase and responsible for reviewing and merging contributions.
+
+In order to propose a change to the codebase, you must open a Pull Request using the GitHub Pull Request feature.
```
Comment (frankieroberto) [inline CONTRIBUTING.md]:
```suggestion
In order to propose a change to the codebase, you can open a pull request on GitHub.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Opening Pull Requests
+1. Try to only open one pull request at a time. Address feedback on your pull request before working on other changes.
+1. If you are a member of the product team, add a link to the relevant team Trello card at the top of the description, if applicable, to give the reader context of the change.
```
Comment (frankieroberto) [inline CONTRIBUTING.md]:
```suggestion
1. If you are a member of the product team, add a link to the relevant team Trello card at the top of the description, if applicable, to give the reader the context of the change.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+1. Only merge your changes once the relevant approvals have been gathered and recorded on the pull request.
+1. The maintainers may close your pull request if it does not meet the expected quality, if feedback is not addressed, or if it does not align with the product team's goals.
+
+## Style guide
+1. Prioritise readability.
+1. The minitest suite is deprecated - use RSpec for new tests, and remove redundant tests from the old test suite when possible.
+1. All new code must be supported by a feature spec and relevant unit specs. Write a new feature spec if one does not already exist.
+1. Feature specs should only cover "happy path" scenarios and some basic error states. Edge cases and exhaustive error testing should be performed in the relevant unit tests.
+1. Use full paths in feature specs - not Rails URL helpers - in order to test the full stack from the user's perspective.
+1. Use services to encapsulate business processes, and models for data integrity. Controllers should deal with interactions between the user and services. We use the interactor gem for composing services. Avoid using organizers to reduce complexity.
```
Comment (frankieroberto) [inline CONTRIBUTING.md]:
```suggestion
1. Use services to encapsulate business processes, and models for data integrity. Controllers should deal with interactions between the user and services. We use the [interactor](https://github.com/collectiveidea/interactor) gem for composing services. Avoid using organizers to reduce complexity.
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (slorek) [reply .github/PULL_REQUEST_TEMPLATE.md]: This was a point I noted down from our meeting, I think it came about because of a Devise controller being overloaded which wasn't particularly obvious. Do you think we can word it differently or drop it?

Comment (scruti) [reply .github/PULL_REQUEST_TEMPLATE.md]: We could put the Devise controller case as example in the check itself.

Comment (frankieroberto) [reply .github/PULL_REQUEST_TEMPLATE.md]: I wasn’t clear what "overloaded" meant in this context... 🤔

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -3,16 +3,20 @@
...
+<!--- Put an `x` in all the boxes that apply. Delete items which are not relevant. -->
 ## Checklist:
-<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
-- [ ] Automated checks are passing locally.
-- [ ] CHANGELOG updated if change is worth telling users about.
-### General testing
-- [ ] Test without javascript
+- [ ] Have you documented your changes in the pull request description?
+- [ ] Does the change present any security considerations?
+- [ ] Is any gem functionality overloaded?
```
Comment (scruti) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+### Authors
+1. Respond to change requests/suggestions with a rationale if you disagree.
+1. Make any changes as may be requested by peers.
+1. Only merge your changes once the relevant approvals have been gathered and recorded on the pull request.
+1. The maintainers may close your pull request if it does not meet the expected quality, if feedback is not addressed, or if it does not align with the product team's goals.
+
+## Style guide
+1. Prioritise readability.
+1. The minitest suite is deprecated - use RSpec for new tests, and remove redundant tests from the old test suite when possible.
```
Comment (scruti) [inline CONTRIBUTING.md]: Should we add the same for `slim` --> `erb` ?
